src/coffee/detail.coffee

```diff
@@ -148,18 +161,22 @@ controllers.DetailCtrl = ($scope, $rootScope, $http, $timeout, I18nFactory, mapS
         observation.fruiting = null
       $scope.location.observation = observation
 
-      $http.post urls.add_location, location: $scope.location
-      .success (data)->
-        console.log("ADDED LOCATION", data.id)
-        console.log(data)
-        $scope.location_id = data.id
-        load_location(data.id)
-        mapStateService.removeLoading()
-        $scope.detail_context = "view_location"
-      .error (data)->
-        console.log("ADD LOCATION FAILED")
-        console.log(data)
-        $rootScope.$broadcast "SHOW-MAP"
+      console.log($scope.location)
+
+      params = $scope.paramsForLocation($scope.location)
```

Comment:
💁 I had to convert `$scope.location` into the correct POST params to be sent to `urls.add_location`. I'd rather not do this, but the Node API seemed to be expecting a much different data structure to be passed in. I have the feeling this might be the case for other POST/PUT routes as well…